src/main/java/uk/gov/digital/ho/hocs/client/document/DocumentS3Client.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.digital.ho.hocs.client.document;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.SSEAwsKeyManagementParams;
+import com.amazonaws.util.StringInputStream;
+import com.amazonaws.util.StringUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.UnsupportedEncodingException;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class DocumentS3Client {
+
+    private final String untrustedS3BucketName;
+    private final AmazonS3 s3Client;
+    private static final String META_DATA_LABEL = "originalName";
+    private final String untrustedBucketKMSKey;
+
+    public DocumentS3Client(AmazonS3 s3Client,
+                            @Value("${document.s3.untrusted-bucket-name}") String untrustedBucketName,
+                            @Value("${document.s3.untrusted-bucket-kms-key}") String untrustedBucketKMSKey) {
+        this.s3Client = s3Client;
+        this.untrustedS3BucketName = untrustedBucketName;
+        this.untrustedBucketKMSKey = untrustedBucketKMSKey;
+    }
+
+    public String storeUntrustedDocument(String originalFilename, String formattedDocument) {
+        ObjectMetadata metaData = buildObjectMetadata(originalFilename, formattedDocument);
+        String tempObjectName = getTempObjectName();
+        try {
+            PutObjectRequest uploadRequest = buildPutObjectRequest(formattedDocument, metaData, tempObjectName);
+            s3Client.putObject(uploadRequest);
+        } catch (UnsupportedEncodingException e) {
+            // Unless this code changes, this should never happen
```

Comment:
If for some reason S3 bucket was unavailable, the upload would fail and presumably throw an error which would be caught here?

Reply:
The s3 client would throw an unchecked exception which would propagate up. The SQS message consumer would then put it on the dead letter queue and do a number or retries.